tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py

```diff
@@ -0,0 +1,203 @@
+import unittest
+
+import functools
+import mock
+import numpy
+from operator import mul
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer.functions.connection import convolution_nd
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.utils import conv
+
+
+@testing.parameterize(*testing.product({
+    'dims': [(10,), (10, 8), (10, 8, 6)],
+    'c_contiguous': [True, False],
+    'cover_all': [True, False],
+    'x_dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'W_dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
+class TestConvolutionND(unittest.TestCase):
+
+    def setUp(self):
+        in_channels = 3
+        out_channels = 2
+        ndim = len(self.dims)
+        ksize = (3,) * ndim
+        self.stride = (2,) * ndim
+        self.pad = (1,) * ndim
+
+        W_scale = numpy.sqrt(1. / functools.reduce(mul, ksize, in_channels))
+        W_shape = (out_channels, in_channels) + ksize
+        self.W = numpy.random.normal(0, W_scale, W_shape).astype(self.W_dtype)
+        self.b = numpy.random.uniform(-1, 1, out_channels).astype(self.x_dtype)
+
+        x_shape = (2, 3) + self.dims
+        self.x = numpy.random.uniform(-1, 1, x_shape).astype(self.x_dtype)
+        gy_shape = (2, 2) + tuple(
+            conv.get_conv_outsize(d, k, s, p, cover_all=self.cover_all)
+            for (d, k, s, p) in zip(self.dims, ksize, self.stride, self.pad))
+        self.gy = numpy.random.uniform(-1, 1, gy_shape).astype(self.x_dtype)
+
+        self.check_forward_options = {}
+        self.check_backward_options = {'dtype': numpy.float64}
+        if self.x_dtype == numpy.float16 or self.W_dtype == numpy.float16:
+            self.check_forward_options = {'atol': 5e-4, 'rtol': 5e-3}
+            self.check_backward_options = {
+                'dtype': numpy.float64, 'atol': 5e-4, 'rtol': 5e-3}
+
+    def check_forward_consistency(self, nobias=False, use_cudnn=False):
+        x_cpu = chainer.Variable(self.x)
+        W_cpu = chainer.Variable(self.W)
+        b_cpu = None if nobias else chainer.Variable(self.b)
+        y_cpu = functions.convolution_nd(
+            x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+            use_cudnn=use_cudnn, cover_all=self.cover_all)
+
+        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
+        W_gpu = chainer.Variable(cuda.to_gpu(self.W))
+        b_gpu = None if nobias else chainer.Variable(cuda.to_gpu(self.b))
+        y_gpu = functions.convolution_nd(
+            x_gpu, W_gpu, b_gpu, stride=self.stride, pad=self.pad,
+            use_cudnn=use_cudnn, cover_all=self.cover_all)
+
+        testing.assert_allclose(
+            y_cpu.data, y_gpu.data.get(), **self.check_forward_options)
```

Comment:
Yes, we can remove calling `get()`. http://docs.chainer.org/en/stable/reference/check.html?highlight=testing#chainer.testing.assert_allclose